### PR TITLE
Add No Qualifying Element

### LINKED ISSUE #30: New Rule: Qualifying Element
As per [SCSS Lint](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#qualifyingelement)

### PR DESCRIPTION
**Pull request by bgriffith**
Rule `no-qualifying-elements` will enforce that selectors are not allowed to have qualifying elements.

There are three options to allow qualifying elements for class, attributes, and/or ids.

### Warning messages are:

> Qualifying element not allowed for attribute

> Qualifying element not allowed for class

> Qualifying element not allowed for id

Closes #30 

DCO 1.1 Signed-off-by: Ben Griffith gt11687@gmail.com

### CODE REVIEW

**Comment by bgriffith** [comment]
Currently blocked by dependence on #159

**Comment by bgriffith** [comment]
Updated - no longer blocked.

File: lib/rules/no-qualifying-elements.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        if (item.is('class') || item.is('attribute') || item.is('id')) {
+          var previous = selectors.content[i - 1] || false;
+
+          if (previous && previous.is('ident')) {
+            if (!parser.options['allow-element-with-' + item.type]) {
+              result = helpers.addUnique(result, {
+                'ruleId': parser.rule.name,
+                'line': item.start.line,
+                'column': item.start.column,
+                'message': 'Qualifying element not allowed for ' + item.type,
```
Comment (DanPurdy) [inline lib/rules/no-qualifying-elements.js]: Should this be plural? `qualifying elements`

**Comment by bgriffith** [comment]
Improved the warning message all round.

**Comment by DanPurdy** [comment]
:+1:

**Pull request merged**